app/server/utils/rewards/processor.py

```diff
@@ -0,0 +1,182 @@
+
+    for counter, (user_id,
+                  user_total_unique_outward_txs) in enumerate(daily_bonus_query_result):
+        # compute total unique outward transactions for all users
+        collective_unique_outward_transactions += user_total_unique_outward_txs
+
+        # convert list into python objects
+        user_id_and_total_unique_outward_transactions_tuple = (user_id, user_total_unique_outward_txs)
+
+        processed_user_ids_with_total_unique_outward_transactions.append(
+            user_id_and_total_unique_outward_transactions_tuple)
+
+        # find user by user id
+        user = get_user_by_id(user_id)
+
+        # define user transfer account
+        user_transfer_account = user.default_transfer_account
+
+        # get user's token
+        user_token = get_transfer_account_token(user_transfer_account)
+
+        # get disbursing admin
+        disbursing_admin = get_admin_account_to_disburse(user)
+
+        # add tuple of user and corresponding total unique outward transaction to list
+        user_and_total_unique_outward_transactions_list.append((user,
+                                                                user_total_unique_outward_txs,
+                                                                disbursing_admin,
+                                                                user_token))
+
+    # persist reward data
+    recipient_data = {
+        'user_ids_with_total_unique_outward_txs': processed_user_ids_with_total_unique_outward_transactions,
+        'collective_unique_outward_txs': collective_unique_outward_transactions
+    }
+
+    persist_reward_data(recipients_data=recipient_data, tag='Daily bonus')
+
+    return user_and_total_unique_outward_transactions_list, collective_unique_outward_transactions
+
+
+def make_daily_bonus_disbursements(created_since: int, issuable_amount: int, transfer_amount: int):
+    daily_bonus_query_result = users_with_corresponding_total_outward_txs(created_since, transfer_amount)
+
+    # process daily bonus data
+    user_and_total_unique_outward_txs_list, collective_unique_outward_txs = process_daily_bonus_query(
+        daily_bonus_query_result=daily_bonus_query_result)
+
+    # iterate through list to make disbursements
+    for counter, \
+        (user,
+         unique_outward_transactions_per_user,
+         disbursing_admin,
+         user_token) in enumerate(user_and_total_unique_outward_txs_list):
+        # compute user's total unique outward transactions as a percentage of the collective unique outward
+        # transactions
+        user_percentage_unique_outward_transactions = (
+            (unique_outward_transactions_per_user / collective_unique_outward_txs))
+
+        # get user's share of the total issuable amount
+        user_bonus_amount = int(user_percentage_unique_outward_transactions * issuable_amount)
+
+        if user_bonus_amount > 1:
+            disbursement = make_payment_transfer(user_bonus_amount,
```

Comment:
Is this an asyncronous call? If not, we could potentially wait a loooong time on this.